addons/account/tests/test_reconciliation_matching_rules.py

```diff
@@ -985,3 +986,105 @@ def create_payment_line(amount, memo, partner):
+                    'tag_ids': [(6, 0, tax_tag_neg.ids)],#+G1,-GST
+                }),
+            ],
+            'refund_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                    'tag_ids': [(6, 0, tax_tag_neg.ids)],#-G1
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, tax_tag_pos.ids)],#-G1,+GST
+                }),
+            ],
+        })
+        reconciliation_model = self.env['account.reconcile.model'].create({
+            'name': 'Charge with Tax',
+            'company_id': self.company.id,
+            'line_ids': [(0, 0, {
+                'company_id': self.company.id,
+                'account_id': self.company_data['default_account_expense'].id,
+                'amount_type': 'percentage',
+                'amount_string': '100',
+                'tax_ids': [(6, 0, [tax.id])]
+            })]
+        })
+        bank_stmt = self.env['account.bank.statement'].create({
+            'company_id': self.company.id,
+            'journal_id': self.bank_journal.id,
+            'date': time.strftime('%Y-07-15'),
+            'name': 'test',
+            'line_ids': [(0, 0, {
+                'payment_ref': 'testLine',
+                'amount': 5,
+                'date': time.strftime('%Y-07-15'),
+                'partner_name': 'test',
+            })]
+        })
+        res = reconciliation_model._get_write_off_move_lines_dict(bank_stmt.line_ids[0], 5)
+        model = [
+            {
+                'name': 'testLine',
+                'balance': 4.55,
+                'debit': 4.55,
+                'credit': 0,
+                'account_id': self.company_data['default_account_expense'].id,
+                'currency_id': False,
+                'analytic_account_id': False,
+                'analytic_tag_ids': [(6, 0, [])],
+                'reconcile_model_id': reconciliation_model.id,
+                'tax_ids': [(6, None, tax.ids)],
+                'tax_tag_ids': [(6, 0, tax.refund_repartition_line_ids[0].tag_ids.ids)]
+            },
+            {
+                'account_id': self.company_data['default_account_expense'].id,
+                'name': tax.name,
+                'partner_id': None,
+                'balance': 0.45,
+                'debit': 0.45,
+                'credit': 0,
+                'analytic_account_id': False,
+                'analytic_tag_ids': False,
+                'tax_exigible': 'on_invoice',
+                'tax_repartition_line_id': tax.invoice_repartition_line_ids[1].id,
+                'tax_ids': [(6, 0, [])],
+                'tax_tag_ids': [(6, 0, tax.refund_repartition_line_ids[1].tag_ids.ids)],
+                'currency_id': False,
+                'reconcile_model_id': reconciliation_model.id,
```

Comment:
you can remove `currency_id`, `debit`, `credit`, `analytic_account_id`, `analytic_tag_ids`, `tax_exigible`
Also, plz use the same order in both dictionaries ><